src/CoreWCF.Kafka/src/CoreWCF/Channels/KafkaTransportPump.cs

```diff
@@ -0,0 +1,246 @@
+
+        ConsumerConfig = TransportBindingElement.Config;
+        ConsumerConfig.BootstrapServers = _baseAddress.Authority;
+        var (enableAutoCommit, enableAutoOffsetStore) = GetCommitStrategyConfigValues(ConsumerConfig, _kafkaDeliverySemantics);
+        ConsumerConfig.EnableAutoCommit = enableAutoCommit;
+        ConsumerConfig.EnableAutoOffsetStore = enableAutoOffsetStore;
+        Consumer = new ConsumerBuilder<Null, byte[]>(ConsumerConfig)
+            .SetKeyDeserializer(Deserializers.Null)
+            .SetValueDeserializer(Deserializers.ByteArray)
+            .SetLogHandler(OnLog)
+            .SetErrorHandler(OnError)
+            .Build();
+
+        Consumer.Subscribe(Topic);
+
+        if (TransportBindingElement.ErrorHandlingStrategy == KafkaErrorHandlingStrategy.DeadLetterQueue)
+        {
+            ProducerConfig producerConfig = new(ConsumerConfig);
+            producerConfig.Acks = Acks.All;
+            Producer = new ProducerBuilder<Null, byte[]>(producerConfig)
+                .SetKeySerializer(Serializers.Null)
+                .SetValueSerializer(Serializers.ByteArray)
+                .Build();
+        }
+
+        Task.Run(async () =>
+        {
+            while (!ct.IsCancellationRequested)
+            {
+                try
+                {
+                    var consumeResult = Consumer.Consume(ct);
+                    if (ConsumerConfig.EnablePartitionEof == true && consumeResult.IsPartitionEOF)
+                    {
+                        continue;
+                    }
+
+                    _logger.LogInformation("Received message from kafka at {topicPartitionOffset}", consumeResult.TopicPartitionOffset);
+                    if (_kafkaDeliverySemantics == KafkaDeliverySemantics.AtMostOnce)
+                    {
+                        Consumer.Commit(consumeResult);
+                    }
+
+                    await OnConsumeMessage(consumeResult, queueTransportContext);
+
+                }
+                catch (OperationCanceledException)
+                {
+                    break;
+                }
+                catch (ConsumeException e)
+                {
+                    if (e.Error.IsFatal)
+                    {
+                        _logger.LogCritical("Exit consume loop {code} {error}", e.Error.Code, e.Error.Reason);
+                        break;
+                    }
+
+                    _logger.LogError(e, "Consume error {code} {error}", e.Error.Code, e.Error.Reason);
+                }
+                catch (Exception e)
+                {
+                    _logger.LogCritical(e, "Unexpected error");
+                    break;
+                }
+            }
+            _mres.Set();
+        }, ct);
+        return Task.CompletedTask;
+    }
+
+    public override async Task StopPumpAsync(CancellationToken token)
+    {
+        _cts.Cancel();
+        await _mres.WaitAsync(CancellationToken.None);
```

Comment:
This will throw a NRE if you call StopPumpAsync before StartPumpAsync.

Reply:
fixed